lntest/itest/lnd_test.go

```diff
@@ -9811,6 +9869,67 @@ out:
 	cleanupForceClose(t, net, net.Bob, chanPointBob)
 }
 
+// assertForward checks that the client provided has a forwarding event
+// that is followed by a forwarding failure or settle.
+func assertForward(t *harnessTest, success bool,
+	eventType routerrpc.HtlcEvent_EventType,
+	client routerrpc.Router_SubscribeHtlcEventsClient) {
+
+	fwdEvent := assertEventAndType(t, eventType, client)
+	_, ok := fwdEvent.Event.(*routerrpc.HtlcEvent_ForwardEvent)
+	if !ok {
+		t.Fatalf("expected forwarding event, got: %T", fwdEvent.Event)
+	}
+
+	// Get the follow up settle or fail event.
+	resolution := assertEventAndType(t, eventType, client)
+
+	// Check for a settle if we expect a successful forward, otherwise check
+	// for a failure.
+	if success {
+		_, ok = resolution.Event.(*routerrpc.HtlcEvent_SettleEvent)
+	} else {
+		_, ok = resolution.Event.(*routerrpc.HtlcEvent_ForwardFailEvent)
+	}
```

Comment:
This assert function uses a somewhat different style compared to the one in the previous commit. Wouldn't it be possible to unify this? Perhaps define an enum with everything that we can expect and then do: `assert(alice, FORWARD, SETTLE)`

A helper function likes `assertForward` is nice, but it also makes it a little less explicit what the raw expectation is.

Reply:
Flattened this to just use the event count. Only `assertLinkFailure` remains separate because we check additional information on the event.